shell.nix

```diff
@@ -1,78 +1,79 @@
-let
-  sources  = import ./nix/sources.nix;
-  commands = import ./nix/commands.nix;
+{ rosetta ? false }:
+  let
+    sources  = import ./nix/sources.nix;
+    commands = import ./nix/commands.nix;
 
-  nixos    = import sources.nixos    {};
-  darwin   = import sources.darwin   {};
-  unstable = import sources.unstable {};
+    overrides = if rosetta then { system = "x86_64-darwin"; } else {};
 
-  pkgs  = if darwin.stdenv.isDarwin then darwin else nixos;
-  tasks = commands {
-    inherit pkgs;
-    inherit unstable;
-    inherit server-path;
-    inherit server-port;
-  };
+    nixos    = import sources.nixos    overrides;
+    darwin   = import sources.darwin   overrides;
+    unstable = import sources.unstable overrides;
```

Comment:
This is a MUCH nicer solution than a copy/pasted rosetta.nix file

Reply:
Is does mean that Brian and I need to `nix-shell --arg rosetta true` and then run stack with `--no-nix`, but that's the same as what Stack does underneath, so 🤷‍♀️